### PR TITLE
add docs github hook

### PR DESCRIPTION
**Pull request by printfcoder**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"github.com/google/go-github/github"
+	"github.com/micro/cli"
+	"github.com/micro/go-log"
+	"github.com/micro/go-web"
+	"net/http"
```
Comment (asim) [inline main.go]: Separate stdlib packages from external ones.

```
net/http

github.com/some/thing
```

File: main.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+	"strings"
+)
+
+var (
+	secret   []byte
+	buildCMD = "bundle exec jekyll build -d /var/html/docs"
+)
+
+func main() {
+
```
Comment (asim) [inline main.go]: no new line

File: main.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+			secret = []byte(c.String("web_hook_secret"))
+
+			if c.String("jekyll_build_cmd") != "" {
+				buildCMD = c.String("jekyll_build_cmd")
+			}
+		}))
+
+	app.HandleFunc("/web-hook", func(w http.ResponseWriter, r *http.Request) {
+
```
Comment (asim) [inline main.go]: no new line

File: main.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+		_, _ = w.Write([]byte("OK"))
+	})
+
+	if err := app.Run(); err != nil {
+		panic(err)
+	}
+}
+
+func pull() error {
+
```
Comment (asim) [inline main.go]: no new line

File: main.go [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+		return err
+	}
+
+	log.Logf("pull log: \n %s", out)
+	log.Logf("pull done \n")
+	return err
+}
+
+func build() error {
+
```
Comment (asim) [inline main.go]: no new line

File: main.go [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+	"github.com/micro/go-log"
+	"github.com/micro/go-web"
+	"net/http"
+	"os/exec"
+	"strings"
+)
+
+var (
+	secret   []byte
+	buildCMD = "bundle exec jekyll build -d /var/html/docs"
```
Comment (asim) [inline main.go]: make the directory it builds to configurable

File: main.go (hunk 1 above)
Comment (printfcoder) [reply main.go]: fixed

File: main.go (hunk 2 above)
Comment (printfcoder) [reply main.go]: fixed

File: main.go (hunk 3 above)
Comment (printfcoder) [reply main.go]: fixed

File: main.go (hunk 4 above)
Comment (printfcoder) [reply main.go]: fixed

File: main.go (hunk 5 above)
Comment (printfcoder) [reply main.go]: fixed

File: main.go (hunk 6 above)
Comment (printfcoder) [reply main.go]: the option buildCMD is configurable, so I think there is unnecessary to make the build dir configurable separately.

**Comment by printfcoder** [comment]
run the app

```bash
./server --server_address=127.0.0.1:8080 --web_hook_secret=123 --jekyll_build_cmd=bundle\ exec\ jekyll\ build\ -d\ /usr/local/var/html/docs
```

**Pull request closed without merging**